src/fake_useragent/utils.py

```diff
@@ -25,53 +17,24 @@ class BrowserUserAgentData(TypedDict):
     """The usage percentage of the user agent."""
     type: str
     """The device type for this user agent (eg. mobile or desktop)."""
-    device_brand: Union[str, None]
+    device_brand: Optional[str]
     """Brand name for the device (eg. Generic_Android)."""
-    browser: Union[str, None]
+    browser: Optional[str]
     """Browser name for the user agent (eg. Chrome Mobile)."""
     browser_version: str
     """Version of the browser (eg. "100.0.4896.60")."""
     browser_version_major_minor: float
     """Major and minor version of the browser (eg. 100.0)."""
-    os: Union[str, None]
+    os: Optional[str]
     """OS name for the user agent (eg. Android)."""
-    os_version: Union[str, None]
+    os_version: Optional[str]
     """OS version (eg. 10)."""
     platform: str
     """Platform for the user agent (eg. Linux armv81)."""
 
 
-def find_browser_json_path() -> Path:
-    """Find the path to the browsers.json file.
-
-    Returns:
-        Path: Path to the browsers.json file.
-
-    Raises:
-        FakeUserAgentError: If unable to find the file.
-    """
-    try:
-        file_path = ilr.files("fake_useragent.data").joinpath("browsers.jsonl")
-        return Path(str(file_path))
-    except Exception as exc:
-        logger.warning(
-            "Unable to find local data/jsonl file using importlib-resources. Try pkg-resource.",
-            exc_info=exc,
-        )
-        try:
-            from pkg_resources import resource_filename
-
-            return Path(resource_filename("fake_useragent", "data/browsers.jsonl"))
-        except Exception as exc2:
-            logger.warning(
-                "Could not find local data/jsonl file using pkg-resource.",
-                exc_info=exc2,
-            )
-            raise FakeUserAgentError("Could not locate browsers.jsonl file") from exc2
```

Comment:
I haven't looked into this, to be honest, so I won't block the merge (not sure I even can 😆, @melroy89 should have final say here I think), but a cursory look online says that `importlib.resources` is valid since Python 3.7 (and also bundled, so it doesn't add dependencies).

If this is true, why should we deprecate this? Maybe just use the first part and ignore the `pkg_resources` part.

Reply:
Some reasons that I am removing this fallback approach too:

1. Similar to `importlib-resources`, `setuptools` is not a built-in package, so it needs to be specified as `dependencies` too
2. Looks like `pkg_resources` is now considered deprecated as an API, and using it will raises a DeprecationWarning
3. According to https://github.com/fake-useragent/fake-useragent/pull/413#issuecomment-2532740132, it seems `pkgutil` won't fail in any case

If in some cases the `pkgutil` fail to load data, I am happy to revert the removal of the original approaches and added that to a new test case.

Reply:
Sorry, I think my message might've been confusing.

[`importlib.resources`](https://docs.python.org/3.9/library/importlib.html?highlight=importlib#module-importlib.resources) is standard library, so it shouldn't add an extra dependency for Python > 3.7. Since we target >=3.9, we should always have the module available.

My (non-bocking) opinion is to either:
1. Keep the `find_browser_json_path` function in the module and use `importlib.resources` only to find the path to the file. Keep your implementation of the loader with `get_data`, which is faster.
1. Move the `find_browser_json_path` function to the `ua_convert.py` script and use `importlib.resources` only to find the path to the file. Keep your implementation of the loader with `get_data`, which is faster.

Reply:
I vote for the 2nd one basically. Because the reason we use `pkgutil` instead of `Path(__file__, '../../data/browsers.jsonl')`, which is simpler and more straight foreword, is that we can't assume that users install this package normally.

For example, people may use zipimport or custom module loader to import this lib, so you can't really _find the path_ of that file. In the zipimport case, the path would be an invalid one, and calling `.read_text()` would raise an error because that path is something like `.../abc.zip/data/browsers.jsonl`.

But, `ua_convert.py` is only used during build-time. We can ensure that the path is a normal path, so things can be very simple.

Reply:
I would like to point again to the same stackoverflow answer: 

https://stackoverflow.com/questions/6028000/how-to-read-a-static-file-from-inside-a-python-package/58941536#58941536

> 
Bad way #1: using relative paths from a source file

Reply:
Yeah. The reason that `__file__` is not preferred in most case:

> What's wrong with that? The assumption that you have files and subdirectories available is not correct. This approach doesn't work if executing code which is packed in a zip or a wheel, and it may be entirely out of the user's control whether or not your package gets extracted to a filesystem at all.

So we won't use this in anywhere of the `fake_useragent` module.

Reply:
I don't fully agree.

People are using this package even under Windows. You have a hard-coded path that most likely only works under Linux.

Furthermore, users also use [pyinstaller](https://pypi.org/project/pyinstaller/) to and package this module in their own single binary on Windows or MacOS. Are you sure this solution is really cross-platform supported?

Reply:
I really tested platform compatibility carefully because I personally use Windows developing.

In fact, one of the reasons that Python is prefering `pathlib.Path` over `os.path.` alternatives is its cross-platform consistency.

More specifically, I ran `ua_convert.py` in this branch myself, on my Windows machine, it works well if I specified an input file. (But `--download` fails because on Windows it seems a `TemporyFile` can't be opened in write and read mode 😂, that's why I used `TemporyDirectory` in the newly added test case in this PR)

Reply:
To be fair, `ua_convert.py` script is officially not part of the package. Its a helper script to.. well convert the UAs :). And is meant to run via CI/CD, using Linux GitHub Actions in this scenario.